File: service/public/src/main/java/org/mlsk/service/admin/AdminService.java
package org.mlsk.service.admin;

import org.mlsk.service.model.admin.EngineDetailResponse;

import java.math.BigDecimal;
import java.util.List;
import java.util.Optional;

public interface AdminService {

  EngineDetailResponse ping(int engineId);

  List<EngineDetailResponse> pingAll();
}

File: web-ui/src/app/admin/service/admin.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

import { EnginesDetail } from '../model/engines-detail';
import { ServerCaller } from 'src/app/shared/service/server-caller/server-caller.service';

@Injectable({
  providedIn: 'root'
})
export class AdminService extends ServerCaller {

  constructor(httpClient: HttpClient) {
    super(httpClient, 'admin');
  }

  ping(id?: number): Observable<EnginesDetail> {
    const httpParams = new HttpParams();
    if (id) {
      httpParams.append('engineId', id!);
    }
    return this.getAndCatchError('ping', httpParams);
  }
}
